android-core/src/main/java/com/mparticle/internal/MParticleApiClientImpl.java

```diff
@@ -203,28 +207,29 @@ public void fetchConfig(boolean force) throws IOException, MPConfigException {
         }
     }
 
-    public JSONObject fetchAudiences() {
-
-        JSONObject response = null;
+    public void fetchUserAudience(BaseAudienceTask task, long mpId) {
+        JSONObject jsonResponse = null;
         try {
-            Logger.debug("Starting Segment Network request");
-            MPConnection connection = getUrl(Endpoint.AUDIENCE).openConnection();
-            connection.setConnectTimeout(mConfigManager.getConnectionTimeout());
-            connection.setReadTimeout(mConfigManager.getConnectionTimeout());
+            MPConnection connection = getUrl(Endpoint.AUDIENCE,mpId).openConnection();
+            Logger.verbose("Audience API request: \n" + connection.getURL().toString());
             connection.setRequestProperty("User-Agent", mUserAgent);
-
             addMessageSignature(connection, null);
+            connection.setRequestProperty("x-mp-key", mApiKey);
             makeUrlRequest(Endpoint.AUDIENCE, connection, true);
             if (connection.getResponseCode() == HttpURLConnection.HTTP_FORBIDDEN) {
-                Logger.error("Segment call forbidden: is Segmentation enabled for your account?");
+                Logger.error("Audience API call forbidden: is the Audience API enabled for your account?");
+            }
+            jsonResponse = MPUtility.getJsonResponse(connection);
+            Logger.verbose("Audience API response: \n" +connection.getResponseCode()+ "  "+ jsonResponse);
```

Comment:
```suggestion
            Logger.verbose("Audience API response: \n Status code: " +connection.getResponseCode()+ "  JSON response: "+ jsonResponse);
```

Reply:
(or something like this, weird to have the status code without telling them it's a status code)

Reply:
also - what happens if jsonResponse is null?